x-pack/filebeat/input/gcs/scheduler.go

```diff
@@ -239,3 +245,12 @@ func (s *scheduler) addFailedJobs(ctx context.Context, jobs []*job) []*job {
 	}
 	return jobs
 }
+
+func (s *scheduler) isFileSelected(name string) bool {
+	for _, sel := range s.src.FileSelectors {
+		if sel.Regex == nil || sel.Regex.MatchString(name) {
```

Comment:
Is it possible to configure a `nil` in `file_selectors regex`

Reply:
not possible to configure nil, but we can have an empty string. This is just a safety net check.

Reply:
But this line gives a wrong impression that if you [some how] manage to get a `nil` in regex then all file patterns are allowed. Doesn't it?

And what happens when regex is `empty string` , just to understand.

Reply:
if regex is an empty string, it will behave like a no-op.

Reply:
Great.. Then , we can safely remove the `nil check`, can't we?

Reply:
yea

Reply:
This is more on a `readability` point of view for me as a developer.. Just not to get mislead looking at a `== nil` always matches the file selector. :-)

Reply:
removed the check